exonum-java-binding/cryptocurrency-demo/start-cryptocurrency-demo.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+# prints a section header
+function header() {
+    local title=$1
+    local rest="========================================================================"
+    echo
+    echo "===[ ${title} ]${rest:${#title}}"
+    echo
+}
+
+set -eu -o pipefail
+
+header "DETECTING ENVIRONMENT"
+
+# EXONUM_HOME shoudl be set to the root directory of Exonum Java App installed
+if [ -z "${EXONUM_HOME+x}" ];
+then
+    echo "ERROR: EXONUM_HOME variable is empty. Please point the EXONUM_HOME to the root of installed Exonum Java App."
+    exit 1
+fi
+echo "EXONUM_HOME=${EXONUM_HOME}"
+
+EXONUM_JAVA_APP="${EXONUM_HOME}/exonum-java"
+
+# Find the latest version of artifact and build if not found.
+ARTIFACT_PATH="$(find ./target -type f -name exonum-java-binding-cryptocurrency-demo-*-artifact.jar)"
+
+if [ -z "${ARTIFACT_PATH}+x" ];
+then
+    mvn install
+    ARTIFACT_PATH="$(find ./target -type f -name exonum-java-binding-cryptocurrency-demo-*-artifact.jar)"
+fi
+echo "ARTIFACT_PATH=${ARTIFACT_PATH}"
+
+# Prepare the services configuration file
+SERVICES_CONFIG_FILE="services.toml"
+SERVICE_NAME="cryptocurrency-demo"
+echo "[user_services]" > ${SERVICES_CONFIG_FILE}
+echo "${SERVICE_NAME} = '${ARTIFACT_PATH}'" >> ${SERVICES_CONFIG_FILE}
+
+#TODO: Can be removed after ECR-3030
+EJB_LOG_CONFIG_PATH="${EXONUM_HOME}/log4j-fallback.xml"
+echo EJB_LOG_CONFIG_PATH=${EJB_LOG_CONFIG_PATH}
+
+# Find the directory containing libjvm.
+JAVA_HOME="$(java -XshowSettings:properties -version 2>&1 > /dev/null | grep 'java.home' | awk '{print $3}')"
+JAVA_LIB_DIR="$(find ${JAVA_HOME} -type f -name libjvm.\* | xargs -n1 dirname)"
+export LD_LIBRARY_PATH="$JAVA_LIB_DIR"
+echo "LD_LIBRARY_PATH=${LD_LIBRARY_PATH}"
+
```

Comment:
I wonder if we shall require the client to install the application properly:

1. Add it to `PATH`, so that it can be launched without `EXONUM_HOME`
1. Set `LD_LIBRARY_PATH` to point to the directory with libjvm.

It is nice to do everything in the script, but, on the other hand, it allows a half-installed application. What shall we put in a script?

Reply:
The app itself seems capable to detect any required info without `EXONUM_HOME`. It can make it harder to detect in the script whether the app is installed. Like it will fail only at the first execution of the app and we're not able to realize whether it caused by wrong arguments or missing app.

Reply:
Well, it turned out that it is possible to check the availability of a binary via `PATH`, but there is still a problem with `LD_LIBRARY_PATH` on OSX. So I solved this by forcing user providing the location of `libjvm` via specific `LIB_JVM_PATH` variable.